docs/adr/006_cups.md

```diff
@@ -0,0 +1,296 @@
+
+#### c. External HTTPS repository
+
+This option requires most manual intervention.
+
+LoRa Device Provisioning tool is not uploading the credential bundle files to the repository.
+
+'Facade' Azure Function can handle a HTTPS endpoint from where the bundles will be downloaded, even though there is no security check or additional AuthN/AuthZ option provided here.
+
+## CUPS Protocol Implementation
+
+### Flow diagram for credential management scenarios
+
+#### Prerequisites
+
+- LoRa Basics™ Station is configured with a cups.uri, a cups.trust and mandatory cups.cert/key
+- The concentrator device twin should include changes as described in following section
+
+#### Flow
+
+```mermaid
+  sequenceDiagram
+    autonumber
+    Concentrator->>CUPS Server: POST /update-info
+    CUPS Server->>IoT Hub: Retrieve 'Concentrator' twin
+    CUPS Server->>CUPS Server: Verifies 'Concentrator' client certificate
+    Note right of CUPS Server: Logic for client certificate validation to be defined. First version including a 'cupsClientThumbprint' to verify against.
+    alt client certificate not valid
+        CUPS Server->>Concentrator: Return forbidden
+    else client certificate valid
+        alt no cups config available
+            CUPS Server->>Concentrator: Return no updates
+        else cups config available
+            alt different cupsUri
+                alt different cupsCredCrc
+                    CUPS Server->>Facade Function: Retrieve CUPS Credential Blob
+                    Facade Function->>CUPS Server: CUPS Credential Blob
+                    CUPS Server->>Concentrator: Updated cupsUri and cupsCred
+                else equal cupsCredCrc
+                    CUPS Server->>Concentrator: Updated cupsUri
+                end
+            else equal cupsUri
+                alt different cupsCredCrc
+                    CUPS Server->>Facade Function: Retrieve CUPS Credential Blob
+                    Facade Function->>CUPS Server: CUPS Credential Blob
+                    CUPS Server->>Concentrator: Updated cupsCred
+                else equal cupsCredCrc
+                    CUPS Server->>CUPS Server: Checks tcUri and tcCredCrc as per above flow and properly return an update
+                end
+            end
+        end
+    end
+```
+
+### IoT Hub related changes
+
+Two changes are expected for IoT Hub.
+
+First one is related to the ability to associate an Azure storage account with IoT Hub.
+
+One of the possible 'sinks' of where the credential files are going to be stored is the IoT Hub linked Azure Blob Storage.
+
+For this option to work, you must [associate an Azure Storage account to IoT Hub](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-file-upload#associate-an-azure-storage-account-to-iot-hub). Any LoRaWan Starter Kit IoT Hub, should have this option enabled by default, therefore some changes are required in the way the template is deployed.
```

Comment:
I am not sure I understand well this. Based on the doc it looks like this is necessary only if a device need to upload file. As far as i understand we don't need to upload files, as the upload would be carried by the provsiioning tool no`?

Reply:
The idea would be that the provisioning tool is instantiating a device client connection to upload the certificates related to such device.

We can revisit this if you have other ideas

Reply:
As discussed, I have removed any reference to the IoT Hub Storage Account link. Let's keep it simple and use the Azure Function Storage Account

Resolving this conversation, feel free to reopen if needed @Mandur